tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/stomp/StompTest.java

```diff
@@ -1501,6 +1501,32 @@ public void testSubscribeToTopicWithNoLocal() throws Exception {
       conn.disconnect();
    }
 
+   @Test
+   public void testSubscribeToTopicWithNoLocalSendWithStomp() throws Exception {
+      conn.connect(defUser, defPass);
+      subscribeTopic(conn, null, null, null, true, true);
+
+      // send a message on the same connection => it should not be received is noLocal = true on subscribe
+      send(conn, getTopicPrefix() + getTopicName(), null, "Hello World");
+
+      ClientStompFrame frame = conn.receiveFrame(100);
+      assertNull(frame, "No message should have been received since subscription was removed");
```

Comment:
```suggestion
      assertNull(frame, "No message should have been received since subscription specified noLocal");
```
(and place c&p from)